framework/src/modules/chain/components/storage/entities/round_delegates.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const {
+	entities: { BaseEntity },
+	utils: {
+		filterTypes: { NUMBER },
+	},
+} = require('../../../../../components/storage');
+
+const sqlFiles = {
+	create: 'round_delegates/create.sql',
+	delete: 'round_delegates/delete.sql',
+	getRoundDelegates: 'round_delegates/get_round_delegates.sql',
+};
+
+/**
+ * Round
+ * @typedef {Object} Round
+ * @property {string} address
+ * @property {number} amount
+ * @property {string} delegate
+ * @property {number} round
+ */
+
+/**
+ * Round Filters
+ * @typedef {Object} filters.Round
+ * @property {number} [round]
+ * @property {number} [round_eql]
+ * @property {number} [round_ne]
+ * @property {number} [round_gt]
+ * @property {number} [round_gte]
+ * @property {number} [round_lt]
+ * @property {number} [round_lte]
+ * @property {number} [round_in]
+ * @property {json} [delegatePublicKeys]
+ */
+
+class RoundDelegates extends BaseEntity {
+	/**
+	 * Constructor
+	 * @param {BaseAdapter} adapter - Adapter to retrieve the data from
+	 * @param {filters.Round} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.addField('delegatePublicKeys', 'string');
+		this.addField('round', 'number', { filter: NUMBER });
+
+		this.extendDefaultOptions({ sort: '' });
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+
+		this.SQLs = this.loadSQLFiles('dpos', sqlFiles, this.sqlDirectory);
+	}
+
+	async get(round) {
```

Comment:
Is this interface consistent with other entities? I think `get` function returns multiple rows, whereas getOne returns a single element. Maybe we should stick with it?
Also, @nazarhussain, should this be in `chain` component or framework level componenet?

Reply:
@SargeKhan Actually I'd prefer to name this function `getRoundDelegates` because it doesn't return the row but just delegate pks. And So far we don't need a get or getOne method in the entity. I'd prefer to add them when we actually need them. What do you think?

Reply:
Yea, getRoundDelegates is fine too